src/charm.py

```diff
@@ -73,41 +68,27 @@ def _on_update_status(self, event: EventBase) -> None:
             pass
         elif not self.external_idp_provider.is_ready():
             self.unit.status = BlockedStatus("Waiting for relation with Kratos")
-        elif not self._stored.redirect_uri and self._stored.enabled:
+        elif not self.external_idp_provider.get_redirect_uri() and self.config["enabled"]:
             self.unit.status = WaitingStatus("Waiting for Kratos to register provider")
+        elif not self.config["enabled"]:
+            self.unit.status = ActiveStatus("Provider is disabled")
         else:
-            msg = "Provider is ready"
-            if not self._stored.enabled:
-                msg = "Provider is disabled"
-
-            self.unit.status = ActiveStatus(msg)
-
-    def _on_redirect_uri_changed(self, event: RedirectURIChangedEvent) -> None:
-        self._stored.redirect_uri = event.redirect_uri
-        self._on_update_status(event)
+            self.unit.status = ActiveStatus("Provider is ready")
 
     def _get_redirect_uri(self, event: ActionEvent) -> None:
```

Comment:
It also seems like most of the possible behaviors for this hook aren't unit tested for

Reply:
I tried to add some more tests.